packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts

```diff
@@ -0,0 +1,1305 @@
+        invoke: {
+          src: 'detectFaceDistance',
+          onDone: {
+            target: 'checkFaceDistanceBeforeRecording',
+            actions: 'updateFaceDistanceBeforeRecording',
+          },
+        },
+      },
+      checkFaceDistanceBeforeRecording: {
+        after: {
+          0: {
+            target: 'initializeLivenessStream',
+            cond: 'hasEnoughFaceDistanceBeforeRecording',
+          },
+          100: { target: 'detectFaceDistanceBeforeRecording' },
+        },
+      },
+      initializeLivenessStream: {
+        invoke: {
+          src: 'openLivenessStreamConnection',
+          onDone: {
+            target: 'notRecording',
+            actions: [
+              'updateLivenessStreamProvider',
+              'spawnResponseStreamActor',
+            ],
+          },
+        },
+      },
+      notRecording: {
+        initial: 'waitForSessionInfo',
+        states: {
+          waitForSessionInfo: {
+            after: {
+              0: {
+                target: '#livenessMachine.recording',
+                cond: 'hasServerSessionInfo',
+              },
+              100: { target: 'waitForSessionInfo' },
+            },
+          },
+        },
+      },
+      recording: {
+        entry: ['clearErrorState', 'startRecording'],
+        initial: 'ovalDrawing',
+        states: {
+          ovalDrawing: {
+            entry: 'sendTimeoutAfterOvalDrawingDelay',
+            invoke: {
+              src: 'detectInitialFaceAndDrawOval',
+              onDone: {
+                target: 'checkFaceDetected',
+                actions: [
+                  'updateOvalAndFaceDetailsPostDraw',
+                  'sendTimeoutAfterOvalMatchDelay',
+                ],
+              },
+              onError: {
+                target: '#livenessMachine.error',
+                actions: 'updateErrorStateForRuntime',
+              },
+            },
+          },
+          checkFaceDetected: {
+            after: {
+              0: {
+                target: 'checkRecordingStarted',
+                cond: 'hasSingleFace',
+              },
+              100: { target: 'ovalDrawing' },
+            },
+          },
+          checkRecordingStarted: {
+            after: {
+              0: {
+                target: 'ovalMatching',
+                cond: 'hasRecordingStarted',
+                actions: 'updateRecordingStartTimestampMs',
+              },
+              100: { target: 'checkRecordingStarted' },
+            },
+          },
+          // Evaluates face match and moves to checkMatch
+          // which continually checks for match until either timeout or face match
+          ovalMatching: {
+            entry: 'cancelOvalDrawingTimeout',
+            invoke: {
+              src: 'detectFaceAndMatchOval',
+              onDone: {
+                target: 'checkMatch',
+                actions: 'updateFaceDetailsPostMatch',
+              },
+            },
+          },
```

Comment:
Updated lines 351 to 368, from
```
       checkMatch: {
            after: {
              0: {
                target: 'flashFreshnessColors',
                cond: 'hasFaceMatchedInOvalWithMinTime',
                actions: [
                  'updateEndFaceMatch',
                  'setupFlashFreshnessColors',
                  'cancelOvalMatchTimeout',
                  'cancelOvalDrawingTimeout',
                ],
              },
              0.1: {
                target: 'ovalMatching',
                cond: 'hasFaceMatchedInOval',
                actions: 'setFaceMatchTimeAndStartFace',
              },
              1: {
                target: 'ovalMatching',
                cond: 'hasNotFaceMatchedInOval',
              },
            },
          },
```
(Removed `hasFaceMatchedInOvalWithMinTime`. Removed 'hasNotFaceMatchedInOval', as the `1:` part will only be reached if `hasFaceMatchedInOval` evaluates to false.

Reply:
Thanks for adding the comment here 👍